From 317bd2df468a08e7b90ae39aa51f62a60757216b Mon Sep 17 00:00:00 2001
From: Shreenidhi Shedi <sshedi@vmware.com>
Date: Mon, 28 Mar 2022 20:44:31 +0530
Subject: [PATCH] system systemd (rebased for 43.6)

Drop deltas merged upstream (resolved socket/bind rules, modules-load
syslog). Keep Photon-only timedated/resolved/devlog and gnomeclock alias
removal.

Signed-off-by: Shreenidhi Shedi <sshedi@vmware.com>
---
 policy/modules/system/systemd.te | 5 +++++
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/policy/modules/system/systemd.te b/policy/modules/system/systemd.te
index 8d96c4b..c7ef3b5 100644
--- a/policy/modules/system/systemd.te
+++ b/policy/modules/system/systemd.te
@@ -187,7 +187,6 @@
 
 systemd_domain_template(systemd_timedated)
 typeattribute systemd_timedated_t systemd_domain;
-typealias systemd_timedated_t alias gnomeclock_t;
 
 type systemd_timedated_unit_file_t;
 systemd_unit_file(systemd_timedated_unit_file_t)
@@ -1215,6 +1214,7 @@
 
 list_dirs_pattern(systemd_timedated_t, systemd_networkd_var_run_t, systemd_networkd_var_run_t)
 read_files_pattern(systemd_timedated_t, systemd_networkd_var_run_t, systemd_networkd_var_run_t)
+read_files_pattern(systemd_timedated_t, systemd_resolved_var_run_t, systemd_resolved_var_run_t)
 
 kernel_rw_psi(systemd_timedated_t)
 
@@ -1223,6 +1223,7 @@
 corecmd_dontaudit_access_check_bin(systemd_timedated_t)
 
 corenet_tcp_connect_time_port(systemd_timedated_t)
+corenet_udp_bind_generic_node(systemd_timedated_t)
 
 dev_rw_realtime_clock(systemd_timedated_t)
 dev_write_kmsg(systemd_timedated_t)
@@ -1657,6 +1658,8 @@
 allow systemd_resolved_t self:process setcap;
 allow systemd_resolved_t self:tcp_socket { accept listen };
 allow systemd_resolved_t self:unix_dgram_socket create_socket_perms;
+allow systemd_resolved_t devlog_t:lnk_file read;
+allow systemd_resolved_t devlog_t:sock_file write;
 
 manage_dirs_pattern(systemd_resolved_t, systemd_resolved_var_run_t, systemd_resolved_var_run_t)
 manage_files_pattern(systemd_resolved_t, systemd_resolved_var_run_t, systemd_resolved_var_run_t)
